Realm/RLMSyncConfiguration.mm

```diff
@@ -270,32 +285,43 @@ - (BOOL)enableFlexibleSync {
     return make_sync_error(errorClass, @(error.message.c_str()), error.error_code.value(), custom);
 }
 
-static void setDefaults(SyncConfig& config, RLMUser *user) {
-    config.client_resync_mode = ClientResyncMode::Manual;
-    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
-
+- (void)setErrorHandler:(RLMUser *)user {
     RLMSyncManager *manager = [user.app syncManager];
     __weak RLMSyncManager *weakManager = manager;
-    config.error_handler = [weakManager](std::shared_ptr<SyncSession> errored_session, SyncError error) {
+    _config->error_handler = [weakManager, self](std::shared_ptr<SyncSession> errored_session, SyncError error) {
```

Comment:
This is a retain cycle.

Reply:
I tried a couple of things. There does have to be some kind of reference to self in order to use the ivar.
I mainly tried a strong self/weak self kind of thing like we do with the weakManager, but couldn't get it working. I'd pass a weak self into the lamdba, then try to make a strong reference before the weak self could possibly become null.

Is that the right track?